libs/article-parser/src/lib/correlation-score/correlation-score.ts

```diff
@@ -77,16 +111,84 @@ function getParagraphCorrelationScore(
   };
 }
 
+/**
+ * Get the correlation score for the segment of a paragraph with the most matches
+ * Will shorten the paragraph as much as possible while trying to mantain the same correlation score
+ * + or - {@code maintainWithinPercent}
+ */
+function getShortestParagraphCorrelationScore(
+  paragraph: string,
+  impacted: string,
+  recommendation: string,
+  impactedSynonyms: string[],
+  recommendationSynonyms: string[],
+  maintainWithinPercent = cutOffs.maintainScoreWithinPercent
+): ParsedArticleParagraph {
+  function calculatePercentageDifference(x: number, y: number): number {
+    return Math.abs((x - y) / x) * 100;
+  }
+  const sentences = paragraph.split('.');
+  // remove the last element if it is empty
+  if (!sentences[sentences.length - 1]) {
+    sentences.pop();
+  }
+  const initScore = getWholeParagraphCorrelationScore(
+    paragraph,
+    impacted,
+    recommendation,
+    impactedSynonyms,
+    recommendationSynonyms
+  ).correlationScore;
+  let currentScore = initScore;
+  let leftInd = 0;
+  let rightIndNonInclusive = sentences.length;
+  while (
+    calculatePercentageDifference(initScore, currentScore) <=
+      maintainWithinPercent / 2 &&
```

Comment:
Why /2? 
If we say maintain with 10% for example, that means we are okay with the score being 10% higher or 10% lower right? So as long as the percentagedifference <= 10 it should be acceptable?

Reply:
My thinking was that for the first loop (which tries to shorten from the end of the paragraph), it would have an allowance half of the given maintain percent. Thus, the other half can come from shortening on the left hand side

Reply:
Ah thanks for the clarification! 
Not something I would recommend changing now, but maybe something to think about in the future: What happens if there's nothing really to remove from the front half but a lot can be removed from the end?

Reply:
Oh yeah, I was thinking that some sort of greedy algorithm could try to optimize it. But I did not want to quite add that complexity yet